### PR TITLE
Expose months value mapping in datetime package

### LINKED ISSUE #431: Daterangepicker nextMonth logic is incorrect
### Bug report

- __Package version(s)__: latest
- __Browser and OS versions__: 1.5

https://github.com/palantir/blueprint/blob/master/packages/datetime/src/dateRangePicker.tsx#L401

This code uses `12` for december, but should be `11` since it's zero-based.

Correct code should read:
```
// 0 is january, 11 is december
function getNextMonth([month, year]: DisplayMonth): DisplayMonth {
  return month === 11 ? [0, year + 1] : [month + 1, year];
}

function getPreviousMonth([month, year]: DisplayMonth): DisplayMonth {
  return month === 0 ? [11, year - 1] : [month - 1, year];
}
```

### PR DESCRIPTION
**Pull request by leebyp**
#### Checklist

- [x] Include tests (covers the edge case in the issue as well as general usage)

#### Changes proposed in this pull request:

- Expose months mapping in datetime for more verbose code
- Fixes changing displayMonth when selecting dates across years

#### Reviewers should focus on:

- should`Months` be an enum (could be inconsistent if we make a 1-indexed variant as @cmslewis mentions in the issue)?
- the problem from the issue can be reproduced by selecting December on the left hand calendar, then select a start date in December and an end date in January (the displayed months will change unnecessarily)

### CODE REVIEW

File: packages/datetime/src/common/months.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+// tslint:disable object-literal-sort-keys
+export const Months = {
```
Comment (adidahiya) [inline packages/datetime/src/common/months.ts]: I think this might be better as a `const enum` -- you can even specify exact values for each enum key

Comment (giladgray) [inline packages/datetime/src/common/months.ts]: yeah i expected this'd be an `enum`. hooray typescript!

Comment (leebyp) [inline packages/datetime/src/common/months.ts]: sweet, even better. ill leave it as a normal enum until we need to do the `1-index` variant then, cool?

File: packages/datetime/src/dateRangePicker.tsx [hunk 2]
```diff
@@ -399,7 +400,7 @@ function getStateChange(value: DateRange,
 type DisplayMonth = [number, number];
 
 function getNextMonth([month, year]: DisplayMonth): DisplayMonth {
-    return month === 12 ? [0, year + 1] : [month + 1, year];
+    return month === Months.DECEMBER ? [0, year + 1] : [month + 1, year];
```
Comment (giladgray) [inline packages/datetime/src/dateRangePicker.tsx]:
```js
[Months.JANUARY, year + 1]
```

**Comment by cmslewis** [comment]
Yesssss. One thing @leebyp - it'd be preferable to do refactors (like introducing the `Months` enum) in a separate PR, rather than bundle codebase-wide refactors into a PR that also changes functionality.

File: packages/datetime/src/dateRangePicker.tsx [hunk 3]
```diff
@@ -399,7 +400,7 @@ function getStateChange(value: DateRange,
 type DisplayMonth = [number, number];
 
 function getNextMonth([month, year]: DisplayMonth): DisplayMonth {
-    return month === 12 ? [0, year + 1] : [month + 1, year];
+    return month === Months.DECEMBER ? [Months.JANUARY, year + 1] : [month + 1, year];
```
Comment (giladgray) [inline packages/datetime/src/dateRangePicker.tsx]: for posterity: this is the bit that fixes #431
